doc/source/ray-core/examples/batch_prediction.ipynb

```diff
@@ -0,0 +1,337 @@
+    "\n",
+    "Let’s take NYC taxi data in 2009 for example. Suppose we have this simple model:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "\n",
+    "def load_model():\n",
+    "    # A dummy model.\n",
+    "    def model(batch: pd.DataFrame) -> pd.DataFrame:\n",
+    "        # Dummy payload so copying the model will actually copy some data\n",
+    "        # across nodes.\n",
+    "        model.payload = np.zeros(100_000_000)\n",
+    "        return pd.DataFrame({\"score\": batch[\"passenger_count\"] % 2 == 0})\n",
+    "    \n",
+    "    return model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The dataset has 12 files (one for each month) so we can naturally have each Ray task to take one file. By taking the model and a shard of input dataset (i.e. a single file), we can define a Ray remote task for prediction:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pyarrow.parquet as pq\n",
+    "import ray\n",
+    "\n",
+    "@ray.remote\n",
+    "def make_prediction(model, shard_path):\n",
+    "    df = pq.read_table(shard_path).to_pandas()\n",
+    "    result = model(df)\n",
+    "\n",
+    "    # Write out the prediction result.\n",
+    "    # NOTE: unless the driver will have to further process the\n",
+    "    # result (other than simply writing out to storage system),\n",
+    "    # writing out at remote task is recommended, as it can avoid\n",
+    "    # congesting or overloading the driver.\n",
+    "    # ...\n",
+    "\n",
+    "    # Here we just return the size about the result in this example.\n",
+    "    return len(result)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The driver launches all tasks for the entire input dataset. In order to not overload cluster and cause OOM, we use ray.wait() to limit the number of in-flight tasks, see details about this design pattern in [Using ray.wait() to limit tasks](https://docs.ray.io/en/master/ray-core/tasks/patterns/limit-tasks.html)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# 12 files, one for each remote task.\n",
+    "input_files = [\n",
+    "        f\"s3://anonymous@air-example-data/ursa-labs-taxi-data/downsampled_2009_full_year_data.parquet\"\n",
+    "        f\"/fe41422b01c04169af2a65a83b753e0f_{i:06d}.parquet\"\n",
+    "        for i in range(12)\n",
+    "]\n",
+    "\n",
+    "# ray.put() the model just once to local object store, and then pass the\n",
+    "# reference to the remote tasks.\n",
+    "model = load_model()\n",
+    "model_ref = ray.put(model)\n",
+    "\n",
+    "# Allow at most 4 tasks in flight.\n",
```

Comment:
Normally we don't need to use `ray.wait()` to limit in-flight tasks if the number of tasks is not too crazy (that causes OOM for task queue). My worry is that people will just copy paste this and use 4 regardless of their cluster size and number of tasks.

For the example, can we just launch all 12 tasks at once. We can add a tip saying if they have too many tasks and experience OOM then they can refer to the ray.wait design pattern.

Reply:
In general, setting a fixed number doesn't work well with autoscaling cluster. How about we change the value from 4 to sys.maxsize as you indicated in the note?

Reply:
SG, done.